irr/include/IEventReceiver.h

```diff
@@ -543,8 +543,9 @@ struct SEvent
 	};
 
 	SEvent() {
-		// would be left uninitialized in many places otherwise
-		MouseInput.Simulated = false;
+		EventType = static_cast<EEVENT_TYPE>(0);
+		// zero the biggest union member we have, which clears all others too
```

Comment:
Would a static assert be possible here?

e.g.
```C++
static_assert(sizeof(*this) - sizeof(EventType) == sizeof(JoystickEvent));
```

Reply:
that wouldn't account for padding